### ISSUE TITLE
Bitmap & Icongroup crashing

### ISSUE THREAD

**Issue by triumphan** [dblock/resourcelib#29]
BitmapResource bitmapResource = new BitmapResource();
            using (ResourceInfo targetFilenameResources = new ResourceInfo())
            {
                targetFilenameResources.Load("explorer.exe");
                Resource existingBitmapResource = targetFilenameResources[Kernel32.ResourceTypes.RT_BITMAP][0];
                bitmapResource.Name = existingBitmapResource.Name;
                bitmapResource.Language = existingBitmapResource.Language;
                Console.WriteLine("Replacing {0}", bitmapResource.Name);
            }

            BitmapFile bitmapFile = new BitmapFile("1.bmp");
            bitmapResource.Bitmap = bitmapFile.Bitmap;
            bitmapResource.Language = ResourceUtil.USENGLISHLANGID;
            Console.WriteLine("DIB: {0}x{1}", bitmapResource.Bitmap.Image.Width, bitmapResource.Bitmap.Image.Height);
            bitmapResource.SaveTo("explorer.exe");

I get this error only when I try to change resources of system fiiles like explorer.exe, shell32.dll, imageres.dll...

System.ComponentModel.Win32Exception: The request is not supported
   at Vestris.ResourceLib.Resource.SaveTo(String filename, ResourceId type, Reso
urceId name, UInt16 lang, Byte[] data))

**Comment by dblock** [comment]
This should work, try to build a test case for this.

**Comment by triumphan** [comment]
No, it doesn't work. I think is because the .MUI file has to be removed before adding / replacing resources. By the way, do you have any idea of how to add a .BIN file to a PE?

**Comment by triumphan** [comment]
I meant MUI directory which includes a .BIN file inside.

**Comment by dblock** [comment]
This is a bit beyond my knowledge of MUI :) Care to explain? Something ResourceLib should do? (feel free to contribute).

**Comment by triumphan** [comment]
MUI is a directory resource which almost all Windows System files like explorer.exe, imageres.dll.. contains. This is an important resource because there is which is stored the language of those system fileMUI is a directory resource which almost all Windows System files like explorer.exe, imageres.dll.. contains. This is an important resource because there is stored the language of those system files (1033) If there isn't a Mui Dir ( and inside the .BIN file) Windows won't be able to load those important files. If you delete the MUI dir of Imageres.dll and then you restart the explorer, the system won't be able to display the folder icons, programs icons and even won't be able to execute programs. So to update resources in a EXE/DLL with a MUI resource you have to delete the MUI dir (save the .BIN File to stream before) , update the resources and then restore the MUI dir and the appropiate .BIN File. I think that's why ResourceLib is not able to update Windows System files resources. 

Sorry for my english, I'm not a native english speaker .

**Comment by triumphan** [comment]
Ups, I pasted some text accidentally. Sorry haha

**Comment by LesFerch** [comment]
@triumphan I realize this goes back 10 years, but did you actually write code to implement the workaround you described?
